src/main/java/io/supertokens/userroles/UserRoles.java

```diff
@@ -63,25 +63,31 @@ public static boolean doesRoleExist(Main main, String role) throws StorageQueryE
     }
 
     // remove a role mapped to a user, if the role doesn't exist throw a UNKNOWN_ROLE_EXCEPTION error
-//    public static void deleteUserRole(Main main, String userId, String role)
-//            throws StorageQueryException, StorageTransactionLogicException {
-//
-//        UserRolesSQLStorage storage = StorageLayer.getUserRolesStorage(main);
-//
-//        storage.startTransaction(con -> {
-//
-//            boolean doesRoleExist = storage.doesRoleExist_Transaction(con, role);
-//
-//            if (doesRoleExist) {
-//                storage.deleteRoleForUser_Transaction(con, userId, role);
-//            } else {
-//                throw new UnknownRoleException();
-//            }
-//
-//            return null;
-//        });
-//    }
-//
+    public static void removeUserRole(Main main, String userId, String role)
+            throws StorageQueryException, StorageTransactionLogicException, UnknownRoleException {
+
+        UserRolesSQLStorage storage = StorageLayer.getUserRolesStorage(main);
+
+        try {
+            storage.startTransaction(con -> {
+
+                boolean doesRoleExist = storage.doesRoleExist_Transaction(con, role);
+
+                if (doesRoleExist) {
+                    storage.deleteRoleForUser_Transaction(con, userId, role);
+                } else {
+                    throw new StorageTransactionLogicException(new UnknownRoleException());
+                }
+                return null;
+            });
+        } catch (StorageTransactionLogicException e) {
+            if (e.actualException instanceof UnknownRoleException) {
+                throw (UnknownRoleException) e.actualException;
+            }
+            throw e;
+        }
```

Comment:
You do not need to start a transaction.. you can just call `deleteRoleForUser_Transaction` and if it returns `false`, you can throw `UnknownRoleException`, else just return.

So you can make it just `deleteRoleForUser` (will require change to plugin interface). How are we doing this (or plan to do this) for removing role and permission - discuss on a call

Reply:
> You do not need to start a transaction.. you can just call deleteRoleForUser_Transaction and if it returns false, you can throw UnknownRoleException, else just return.

Hey this works for the case where the role doesn't exist, but, if the role exists but it hasn't been mapped to a user, this will still throw an `UNKNOWN_ROLE_EXCEPTION`